### PR TITLE
Use image outputs in examples

### PR DESCRIPTION
**Pull request by bobcatfish**
While talking with @jonjohnsonjr about #216 I wanted to show him an
example of a Pipeline wherein there are Tasks that output Images
Resources, which are then used as inputs for subsequent Tasks. Our
examples Pipelines were meant to do that, but unfortunately when I
looked at them, I realized these were not actually hooked up properly.
The Tasks were not actually using the Images that were built by previous
Tasks, so I have updated them to do that.

I also discovered that the templating was incorrect (#108 - adding tests
for these examples - and #212 - making sure templated variables are
actually used - can't come quickly enough!) so I've updated that in the
examples as well.

### CODE REVIEW

**Comment by pivotal-nader-ziada** [comment]
@bobcatfish everything looks fine to me in the yaml examples. but the test is failing for what seems to be an infrastructure issue

**Comment by bobcatfish** [comment]
I think there are 2 problems, 1 is an infrastructure problem, the other is that the k8s clusters were upgraded to 1.11 and I think we're running into something similar to https://github.com/kubevirt/kubevirt/issues/1279

**Comment by adrcunha** [comment]
Since 5AM all CI flows are failing when creating a k8s cluster. I'll investigate.

**Comment by bobcatfish** [comment]
Thanks @adrcunha  - looks like https://status.cloud.google.com//incident/compute/18012?_ga=2.196962540.-1744253383.1524667138 is over now, so we'll maybe be back to the 1.11 issue which I can look into:

/test pull-knative-build-pipeline-integration-tests

File: examples/deploy_tasks.yaml [hunk 1]
```diff
@@ -44,7 +53,27 @@ spec:
...
         steps:
+        - name: replaceRedisImage
+          image: busybox
+          command: ['sed']
+          args:
+          - "-ri"
+          - "'s/image: k8s.gcr.io\/redis:e2e/image: ${inputs.resources.redisImage.url}@{inputs.resources.redisImage.digest}/' ${inputs.params.pathToFiles}"
+        - name: replaceGuestbookImage
+          image: busybox
+          command: ['sed']
```
Comment (bobcatfish) [inline examples/deploy_tasks.yaml]: feedback from @jonjohnsonjr : I'm going to at least TRY using kustomize before I resign myself to `sed` here :D

**Comment by bobcatfish** [comment]
Should be fixed by #226 :crossed_fingers: 

/test pull-knative-build-pipeline-integration-tests

File: examples/test_tasks.yaml [hunk 2]
```diff
@@ -34,9 +34,8 @@ spec:
         resources:
            - name: workspace
              type: git
+           - name: testImage
```
Comment (pivotal-nader-ziada) [inline examples/test_tasks.yaml]: the test is failing because the `testImage` here needs a type of `image`

**Comment by bobcatfish** [comment]
PTAL @shashwathi @pivotal-nader-ziada

File: examples/deploy_tasks.yaml [hunk 3]
```diff
@@ -44,7 +53,27 @@ spec:
...
+          image: lachlanevenson/k8s-kubectl
           command: ['kubectl']
-          args: ['--use-context', '${targetCluster}', '--namespace', '${targetCluster.namespace}', 'apply', '-c', '${pathToFiles}', '${kubectlArgs}']
+          args:
+          - '--use-context'
+          - '${inputs.clusters.targetCluster}'
+          - '--namespace'
+          - '${inputs.clusters.targetCluster.namespace}'
+          - 'apply'
+          - '-c'
```
Comment (pivotal-nader-ziada) [inline examples/deploy_tasks.yaml]: should this `-c` be `-f`?

File: examples/pipelines/kritis-resources.yaml [hunk 4]
```diff
@@ -10,26 +10,13 @@ spec:
...
-  namespace: default
-spec:    
-  type: git
-  params:
-  - name: revision
-    value: master
-  - name: url
-    value: https://github.com/grafeas/kritis-test  
----    
+---
```
Comment (pivotal-nader-ziada) [inline examples/pipelines/kritis-resources.yaml]: There are use cases when having tests in a separate repo is valid, for example having acceptance test that make sure products from different repos integrate and deploy properly. Anyways doesn't have to be part of our example, but it was good way to show using multiple git repos in the same task

File: examples/test_tasks.yaml [hunk 5]
```diff
@@ -62,10 +62,10 @@ spec:
         volumes:
             - name: gac
               hostPath:
-                path: ${workspace}/config/gac.json
+                path: /workspace/config/gac.json
             - name: cloudconfig
               hostPath:
-                path: ${workspace}/config/gcloud
+                path: /workspace/config/gcloud
```
Comment (pivotal-nader-ziada) [inline examples/test_tasks.yaml]: the path here refers to `/workspace` while the volume mount  int the step refers to `/root`

File: examples/build_task.yaml [hunk 6]
```diff
@@ -23,5 +23,5 @@ spec:
       command:
       - /kaniko/executor
       args:
-      - --dockerfile=${pathToDockerFile}
-      - --destination=$builtImage
+      - --dockerfile=${inputs.resources.pathToDockerFile}
```
Comment (pivotal-nader-ziada) [inline examples/build_task.yaml]: this should be `inputs.params`

File: examples/deploy_tasks.yaml (hunk 3 above)
Comment (bobcatfish) [reply examples/deploy_tasks.yaml]: whoops!

File: examples/pipelines/kritis-resources.yaml (hunk 4 above)
Comment (pivotal-nader-ziada) [reply examples/pipelines/kritis-resources.yaml]: I’m okay with your change :)

File: examples/test_tasks.yaml (hunk 5 above)
Comment (bobcatfish) [reply examples/test_tasks.yaml]: hm I think that's okay, the docs for `hostPath` (here) say:

> HostPath represents a pre-existing file or directory on the host machine that is directly exposed to the container. 

So I think what this is doing is creating a volume from a dir that exists already on the host (which was mounted into `/workspace` via the resource), then in volume mounts it's mounting this dir into a specific location in `/root`

File: examples/build_task.yaml (hunk 6 above)
Comment (bobcatfish) [reply examples/build_task.yaml]: good catch!

**Pull request merged**